target/arm/arm-semi.c

```diff
@@ -303,13 +300,7 @@ static target_ulong arm_flen_buf(ARMCPU *cpu)
      * We put this on the guest's stack just below SP.
      */
     CPUARMState *env = &cpu->env;
-    target_ulong sp;
-
-    if (is_a64(env)) {
-        sp = env->xregs[31];
-    } else {
-        sp = env->regs[13];
-    }
+    target_ulong sp = arm_get_xreg(env, 31);
```

Comment:
Isn't this getting the wrong register `31` vs `13` on from the else branch?

Reply:
Yes, missed that it was different on the different branches.